fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/util/OxidizableFamily.java

```diff
@@ -0,0 +1,192 @@
+ * Unaffected, Exposed, Weathered, and Oxidized.
+ *
+ * <p>Each block also has a 'Waxed' form.
+ */
+public record OxidizableFamily(
+		ImmutableMap<OxidizationLevel, Block> unwaxed,
+		ImmutableMap<OxidizationLevel, Block> waxed) {
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	/**
+	 * Get the unwaxed variant in this family for the given {@link OxidizationLevel}.
+	 * @param level the {@link OxidizationLevel}
+	 * @return the unwaxed variant
+	 */
+	public Block unwaxed(OxidizationLevel level) {
+		return unwaxed().get(level);
+	}
+
+	/**
+	 * Get the waxed variant in this family for the given {@link OxidizationLevel}.
+	 * @param level the {@link OxidizationLevel}
+	 * @return the waxed variant
+	 */
+	public Block waxed(OxidizationLevel level) {
+		return waxed().get(level);
+	}
+
+	/**
+	 * Builds a map of {@link OxidizationLevel} increases.
+	 * @return the map
+	 */
+	public BiMap<Block, Block> oxidizationLevelIncreasesMap() {
+		return ImmutableBiMap.<Block, Block>builder()
+				.put(unwaxed(OxidizationLevel.UNAFFECTED), unwaxed(OxidizationLevel.EXPOSED))
+				.put(unwaxed(OxidizationLevel.EXPOSED), unwaxed(OxidizationLevel.WEATHERED))
+				.put(unwaxed(OxidizationLevel.WEATHERED), unwaxed(OxidizationLevel.OXIDIZED))
+				.build();
+	}
+
+	/**
+	 * Builds a map of {@link OxidizationLevel} decreases.
+	 * @return the map
+	 */
+	public BiMap<Block, Block> oxidizationLevelDecreasesMap() {
+		return oxidizationLevelIncreasesMap().inverse();
+	}
+
+	/**
+	 * Builds a map of unwaxed forms to waxed counterparts.
+	 * @return the map
+	 */
+	public BiMap<Block, Block> unwaxedToWaxedMap() {
+		return ImmutableBiMap.<Block, Block>builder()
+				.put(unwaxed(OxidizationLevel.UNAFFECTED), waxed(OxidizationLevel.UNAFFECTED))
+				.put(unwaxed(OxidizationLevel.EXPOSED), waxed(OxidizationLevel.EXPOSED))
+				.put(unwaxed(OxidizationLevel.WEATHERED), waxed(OxidizationLevel.WEATHERED))
+				.put(unwaxed(OxidizationLevel.OXIDIZED), waxed(OxidizationLevel.OXIDIZED))
+				.build();
+	}
+
+	/**
+	 * Builds a map of waxed forms to unwaxed counterparts.
+	 * @return the map
+	 */
+	public BiMap<Block, Block> waxedToUnwaxedMap() {
+		return unwaxedToWaxedMap().inverse();
+	}
+
+	/**
+	 * Allows for the creation of {@link OxidizableFamily}s.
+	 */
+	public static class Builder {
+		private final HashMap<OxidizationLevel, Block> unwaxed = new HashMap<>();
+		private final HashMap<OxidizationLevel, Block> waxed = new HashMap<>();
+
+		/**
+		 * Adds blocks to the {@link OxidizableFamily}.
+		 * Will output a warning to the log if the unwaxed variant is not an instance of {@link Oxidizable}
+		 * @param level the {@link OxidizationLevel} of the blocks to add
+		 * @param unwaxed the unwaxed variant
+		 * @param waxed the waxed variant
+		 * @return this builder
+		 */
+		public Builder add(OxidizationLevel level, Block unwaxed, Block waxed) {
+			if (!(unwaxed instanceof Oxidizable)) {
```

Comment:
What is the point of this interface. 🤔

Reply:
What interface? Oxidizable? That's a vanilla interface.